### PR TITLE
ICU-22015 Add U_SUPPORTS_ATOMICS

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22015
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/common/unicode/platform.h [hunk 1]
```diff
@@ -347,6 +347,18 @@
 #   define U_HAVE_INTTYPES_H U_HAVE_STDINT_H
 #endif
 
+/**
+ * \def U_SUPPORTS_ATOMICS
+ * Defines whether the platform supports C++ atomics.
+ * Set this to 0 to disable certain code that requires atomics.
```
Comment (markusicu) [inline icu4c/source/common/unicode/platform.h:353]: I would add another sentence like
Only a small part of the ICU code currently avoids C++ atomics based on this setting.

File: icu4c/source/common/unicode/platform.h [hunk 2]
```diff
@@ -347,6 +347,18 @@
...
+ * @internal
```
Comment (sffc) [inline icu4c/source/common/unicode/platform.h:354]: Should it be `@internal`? It seems that many of the other flags are `@internal`, even if the have a `U_` prefix (not just `UPRV_`)

Comment (markusicu) [reply icu4c/source/common/unicode/platform.h:354]: good idea, especially since it's not used everywhere so you can't build all of ICU with it

**Comment by markusicu** [comment]
FYI @aheninger

**Comment by markusicu** [comment]
https://github.com/unicode-org/icu/pull/2080 might be a simpler solution for what Shane is trying to do.

**Comment by aheninger** [comment]
I pretty strongly prefer just doing #2080, and withdrawing this PR.

There's no clear definition of what ICU services U_SUPPORTS_ATOMICS would enable or disable, and trying to tie it down would be hard.

**Comment by sffc** [comment]
Sounds good to me!

**Pull request closed without merging**